modules/ml/src/main/java/org/apache/ignite/ml/svm/SVMLinearBinaryClassificationTrainer.java

```diff
@@ -67,34 +87,32 @@
 
     /** */
     @NotNull private Vector initializeWeightsWithZeros(int vectorSize) {
-        if (isDistributed)
-            return new SparseDistributedVector(vectorSize);
-        else
             return new DenseLocalOnHeapVector(vectorSize);
     }
 
     /** */
-    private Vector calculateUpdates(LabeledDataset data, Vector weights) {
-        Vector copiedWeights = weights.copy();
-        Vector deltaWeights = initializeWeightsWithZeros(weights.size());
-
-        final int amountOfObservation = data.rowSize();
+    private Vector calculateUpdates(Vector weights) {
+        return dataset.compute(data -> {
+            Vector copiedWeights = weights.copy();
```

Comment:
Looks like we can avoid `copy` call because `weights` are passed via serialization-deserialization procedure anyway, so it's not the same object.